### PR TITLE
[ArtemisNG] Better diff handling

### PR DESCRIPTION
**Pull request by pbench**
Basically https://jira.kisio.org/browse/NAVP-1431 with some cleaning along the way

### CODE REVIEW

File: artemis/base_pytest.py [hunk 1]
```diff
@@ -50,7 +50,12 @@ def before_each_test(self, request):
         Note: py.test does not want to collect class with custom constructor,
         so we init the class in the setup
         """
-        self.test_counter = Counter()
```
Comment (mbounabi) [inline artemis/base_pytest.py]: this variable is removed here and still used

Comment (pbench) [reply artemis/base_pytest.py]: This variable is used only in get_file_name().
9aa0772cf7567e4c95751ce74046128c9986b0d6 moved the function get_file_name() to test_mechanism.py as it is only used in the old artemis.
So the behavior of old artemis does not change.

File: artemis/base_pytest.py [hunk 2]
```diff
@@ -361,26 +357,29 @@ def print_diff(ref_file, resp_file):
...
-                os.makedirs(dir_path)
+            # create the directory if it does not exists
+            if not os.path.exists(output_dir_path):
+                os.makedirs(output_dir_path)
 
-            # create path to ref and resp
-            full_file_name_ref = dir_path + '/reference_' + file_name + '.txt'
-            full_file_name_resp = dir_path + '/response_' + file_name + '.txt'
+            # write response file
+            response_filename = "{}_resp.json".format(filename_prefix)
```
Comment (xlqian) [inline artemis/base_pytest.py]: "{}.json"

File: artemis/base_pytest.py [hunk 3]
```diff
@@ -361,26 +357,29 @@ def print_diff(ref_file, resp_file):
...
+            response_filepath = os.path.join(output_dir_path, response_filename)
+            self.write_full_response_to_file(http_query, response_string, response_filepath, response_checker)
 
-            json_filtered_reference = json.dumps(filtered_reference, indent=4)
-            json_filtered_response = json.dumps(filtered_response, indent=4)
+            # write reference file
+            output_reference_filename = "{}_ref.json".format(filename_prefix)
```
Comment (xlqian) [inline artemis/base_pytest.py]: probably there is no more need to save {}_ref.json

File: artemis/base_pytest.py [hunk 4]
```diff
@@ -361,26 +357,29 @@ def print_diff(ref_file, resp_file):
...
 
-            # Save resp and ref as txt files in folder named outputs
-            ref_resp2files(full_file_name_ref, json_filtered_reference)
-            ref_resp2files(full_file_name_resp, json_filtered_response)
+            with open(output_reference_filepath, 'w') as reference_text:
+                reference_text.write(raw_reference)
 
             # Print difference in console
-            print_diff(full_file_name_ref, full_file_name_resp)
+            print_diff(response_filepath, output_reference_filepath, self.get_test_name())
```
Comment (xlqian) [inline artemis/base_pytest.py]: remove print_diff if possible

File: artemis/common_fixture.py [hunk 5]
```diff
@@ -47,30 +48,39 @@ def clean_kirin_db():
...
+        if self.nb_call_to_request_compare <= 1:
+            return func_name
+        else :
+            assert self.nb_call_to_request_compare > 1
+            return "{}_{}".format(func_name, self.nb_call_to_request_compare - 1)
+
+    def get_test_name(self):
+        path = os.path.join(self.get_reference_suffix_path(),
+                            self.get_reference_filename_prefix())
+        return str(path)
```
Comment (xlqian) [inline artemis/common_fixture.py]: we remove this function if we no longer need print_diff

File: artemis/base_pytest.py [hunk 6]
```diff
@@ -275,45 +281,38 @@ def journey(self, _from, to, datetime,
...
 
-        def ref_resp2files(output_file, output_json):
-            """
-            Create a file for the filtered response and for the filtered reference
-            """
-            with open(output_file, 'w') as reference_text:
-                reference_text.write(output_json)
-
-        def print_diff(ref_file, resp_file):
+        def print_diff(ref_file, resp_file, test_name):
```
[the remark this answers is not in the document]
Comment (mbounabi) [reply artemis/base_pytest.py]: maybe add an option to print it?

**Pull request merged**